support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -0,0 +1,265 @@
+		break;
+	case 'au':
+		currentCurrency = currencies.AUD;
+		currentCurrencyKey = 'AUD';
+		break;
+	case 'eu':
+		currentCurrency = currencies.EUR;
+		currentCurrencyKey = 'EUR';
+		break;
+	case 'nz':
+		currentCurrency = currencies.NZD;
+		currentCurrencyKey = 'NZD';
+		break;
+	case 'ca':
+		currentCurrency = currencies.CAD;
+		currentCurrencyKey = 'CAD';
+		break;
+}
+
+const ProductSchema = object({
+	ratePlans: record(
+		object({
+			id: string(),
+			charges: record(
+				object({
+					id: string(),
+					pricing: record(number()),
+				}),
+			),
+		}),
+	),
+});
+
+const ProductsSchema = object({
+	products: record(ProductSchema),
+});
+type Products = Output<typeof ProductsSchema>;
+
+function describeProduct(
+	productId: string,
+	ratePlanId: string,
+	chargeId: string,
+) {
+	let description = `${productId} - ${ratePlanId} ${chargeId}`;
+	let frequency = '';
+
+	if (productId === 'HomeDelivery') {
+		frequency = 'month';
+		description = `${ratePlanId} paper`;
+
+		if (ratePlanId === 'Sixday') {
+			description = 'Six day paper';
+		}
+		if (ratePlanId === 'Everyday') {
+			description = 'Every day paper';
+		}
+		if (ratePlanId === 'Weekend') {
+			description = 'Weekend paper';
+		}
+		if (ratePlanId === 'Saturday') {
+			description = 'Saturday paper';
+		}
+		if (ratePlanId === 'Sunday') {
+			description = 'Sunday paper';
+		}
+	}
+
+	if (
+		productId === 'GuardianWeeklyDomestic' ||
+		productId === 'GuardianWeeklyRestOfWorld'
+	) {
+		if (ratePlanId === 'OneYearGift') {
+			frequency = 'year';
+			description = 'The Guardian Weekly Gift Subscription';
+		}
+		if (ratePlanId === 'Annual') {
+			frequency = 'year';
+			description = 'The Guardian Weekly';
+		}
+		if (ratePlanId === 'Quarterly') {
+			frequency = 'quarter';
+			description = 'The Guardian Weekly';
+		}
+		if (ratePlanId === 'Monthly') {
+			frequency = 'month';
+			description = 'The Guardian Weekly';
+		}
+		if (ratePlanId === 'ThreeMonthGift') {
+			frequency = 'quarter';
+			description = 'The Guardian Weekly Gift Subscription';
+		}
+		if (ratePlanId === 'SixWeekly') {
+			frequency = 'month';
+			description = 'The Guardian Weekly';
+		}
+	}
+
+	return { description, frequency };
+}
+
+export function Checkout() {
+	const [products, setProducts] = useState<Products>();
+
+	useEffect(() => {
+		void fetch('/api/products')
+			.then((resp) => resp.json())
+			.then((data) => {
+				const vData = parse(ProductsSchema, data);
+				setProducts(vData);
+			});
+	}, []);
+
+	if (!query.productId || !query.ratePlanId || !query.chargeId) {
+		return <div>Not enough query parameters</div>;
+	}
+
+	const currentProduct = products?.products[query.productId];
+	const currentRatePlan = currentProduct?.ratePlans[query.ratePlanId];
+	const currentCharge = currentRatePlan?.charges[query.chargeId];
+	const currentPrice = currentCharge?.pricing[currentCurrencyKey] ?? 0;
+
+	if (!currentProduct) {
+		return <div>Product not found</div>;
+	}
+
+	return (
+		<PageScaffold
```

Comment:
I have copy/pasted most of everything from [`secondStepCheckout`](https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/pages/supporter-plus-landing/twoStepPages/secondStepCheckout.tsx) and [`SupporterPlusCheckoutScaffold`](https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/pages/supporter-plus-landing/twoStepPages/checkoutScaffold.tsx#L129).

I have chose to use `PageScaffold` over `SupporterPlusCheckoutScaffold` as that abstraction is full of a lot of `if` statements differentiating between landing and checkout pages, making it feel like a premature abstraction. 

I could go back and refactor that, but it feels like putting effort into this and migrating that to here would be a more efficient process.

Reply:
Yes this is used throughout te code-base including the pre-render S+  page. Makes sense to re-use it